libraries/integration/Microsoft.Bot.Builder.ApplicationInsights.Core/ServiceCollectionExtensions.cs

```diff
@@ -38,10 +37,10 @@ public static IServiceCollection AddBotApplicationInsights(this IServiceCollecti
 
             // Enables Bot Telemetry to save user/session id's as the bot user id and session
             services.AddMemoryCache();
-            services.TryAddTransient<TelemetrySaveBodyASPMiddleware>();
-            services.TryAddSingleton<ITelemetryInitializer>(new OperationCorrelationTelemetryInitializer());
-            services.TryAddSingleton<ITelemetryInitializer, TelemetryBotIdInitializer>();
-            services.TryAddSingleton<IBotTelemetryClient, BotTelemetryClient>();
+            services.AddTransient<TelemetrySaveBodyASPMiddleware>();
```

Comment:
We need to talk about the implementation of these ASP.NET middleware classes, but I guess I need to go back to the old PR to do that because they aren't changed in this PR. 

High level points:
 1) I don't think it's a good idea to do this with ASP.NET middleware _at all_. Core stuff like this should not assume it's even being hosted in ASP.NET.
 2) The current implementation is extremely "hostile" in terms of taking over reading of the body... it will do this for every single (`POST`) request even if it's not targeting the bot itself. Just... no.
 3) The current implementation takes on a dependency of using `MemoryCache` which is completely unnecessary. At bare minimum this should be fixed... you can just stick the data into the `HttpContext::Items` collection and it will flow with the context itself.

I'll go back to the other PR and address this stuff in more detail, but I _implore_ you not to do it this way.

Reply:
Completely agree - there's a lack of a [rewind enablement](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.internal.bufferinghelper.enablerewind?view=aspnetcore-2.1) and once fixed, this component can completely go away..